provider/azure/internal/errorutils/errors.go

```diff
@@ -4,11 +4,19 @@
 package errorutils
 
 import (
+	"net/http"
+
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/azure"
 	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/utils/set"
+
+	"github.com/juju/juju/environs/context"
 )
 
+var logger = loggo.GetLogger("juju.provider.azure")
```

Comment:
I have thought about it but I am on the fence. I'd rather leave it as-is simply because our logging is not generally grouped by package but by functional area and here the functional area is azure provider. SO all the logging for azure provider is either desired or not :)

I have changed the logging of error to Warning although, I do not know what an operator can do with it once the failure occurs.

Greta pickup overall though, as I was not logging the actual failure :) Fixed now.